### PR TITLE
Log remote requests made by the AWS SDK

### LINKED ISSUE #125: Log remote requests made by the AWS SDK
I'm seeing some huge perf issues where being able to dig into which AWS requests are made would be super useful. Because AWS SDK requests go via Guzzle rather than via WP_Http they're not currently logged.

(I know @johnbillion ran into similar things with Query Monitor's HTTP logging, so maybe there's some solution he saw there?)

### PR DESCRIPTION
**Pull request by dashaluna**
Fixes #125

### CODE REVIEW

**Comment by roborourke** [comment]
Tested this out locally with the `audit-log-to-cloudwatch` feature on and the SDK params filter is applied.

@joehoyle @rmccue can you think of any other key things to help test this locally? Can you see any issue with where the filter is?

**Comment by rmccue** [comment]
> @joehoyle @rmccue can you think of any other key things to help test this locally? Can you see any issue with where the filter is?

Let's roll it onto a development stack and test against the live service.

**Comment by roborourke** [comment]
Will update `platform-test` to run this branch 👍

File: load.php [hunk 1]
```diff
@@ -39,3 +40,6 @@
 
 	register_module( 'cloud', __DIR__, 'Cloud', $default_settings, __NAMESPACE__ . '\\bootstrap' );
 } );
+
+// Early hook for logging AWS SDK HTTP requests.
+add_filter( 'altis.aws_sdk.params', __NAMESPACE__ . '\\add_aws_sdk_xray_callback' );
```
Comment (rmccue) [inline load.php]: We need to separately hook into S3 Upload's copy of the SDK currently.

```suggestion
add_filter( 'altis.aws_sdk.params', __NAMESPACE__ . '\\add_aws_sdk_xray_callback' );
add_filter( 's3_uploads_s3_client_params', __NAMESPACE__ . '\\add_aws_sdk_xray_callback' );
```

**Comment by rmccue** [comment]
With the above change, I'm now seeing requests come in, however the data on them appears to be wrong. See Slack for more.

**Comment by rmccue** [comment]
Filed https://github.com/humanmade/aws-xray/issues/54 for the invalid-ish data.

We can move ahead with this in the meantime. With the above change, it should be good to go.

**Comment by rmccue** [comment]
Ah, we probably need to add `aws_ses_wp_mail_ses_client_params` too.

**Comment by roborourke** [comment]
Woot

**Comment by joehoyle** [comment]
Do we need to do anything else for _this_ PR, or shall we merge and add follow-ups for the other items?

**Comment by roborourke** [comment]
Added AWS SES params filter as well.

**Pull request merged**